pkg/api/api.go

```diff
@@ -1736,82 +1759,75 @@ func (s *ScalewayAPI) GetDashboard() (*ScalewayDashboard, error) {
 }
 
 // GetServerID returns exactly one server matching or dies
-func (s *ScalewayAPI) GetServerID(needle string) string {
+func (s *ScalewayAPI) GetServerID(needle string) (string, error) {
 	// Parses optional type prefix, i.e: "server:name" -> "name"
 	_, needle = parseNeedle(needle)
 
 	servers, err := s.ResolveServer(needle)
 	if err != nil {
-		log.Fatalf("Unable to resolve server %s: %s", needle, err)
```

Comment:
From what I remembert, we used `log.Fatalf` here instead of standard error return, because GetServerID may be called in a  goroutine, we need to check that the goroutine will forward the error to a channel or at least move the `fatal` behaviour inside the goroutine, so we can keep this helper fatal-proof as you just did